pkg/operation/botanist/component/nginxingress/nginxingress.go

```diff
@@ -147,6 +148,10 @@ func (n *nginxIngress) computeResourcesData() (map[string][]byte, error) {
 
 	utilruntime.Must(kutil.MakeUnique(configMap))
 
+	k8sVersionGreaterEqual121, err := version.CompareVersions(n.values.KubernetesVersion, ">=", "1.21")
```

Comment:
Can you change `n.values.KubernetesVersion` to `*semver.Version` and use the global version constraints instead?

Reply:
For seed components, we have to use this function because it normalizes the version. Ref: https://github.com/gardener/gardener/pull/6468

Reply:
Well, fair point... Another thing that is not obvious and that we probably should clean up. OK, let's keep it like this then.